contracts/c-sphincs-all-in-one-lock/ckb-sphincsplus.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Glue code between CKB world and sphincsplus.
+ *
+ * Notice the message signed / verified here, is actually preprocessed
+ * message with context prepended. The call is in charge of proper handling
+ * of contexts.
+ */
+
+#include "ckb-sphincsplus.h"
+
+#include <stdint.h>
+
+#include "address.h"
+#include "api.h"
+#include "context.h"
+#include "fors.h"
+#include "params.h"
+#include "thash.h"
+#include "utils.h"
+#include "wotsx1.h"
+
+// #include "ckb_vm_dbg.h"
+
+enum SphincsPlusError {
+  SphincsPlusError_Params = 200,
+  SphincsPlusError_Verify,
+  SphincsPlusError_OutputSignLength,
+};
+
+#ifndef CKB_VM
+
+#include <stdlib.h>
+
+int sphincs_plus_generate_keypair(uint8_t *pk, uint8_t *sk) {
+  return crypto_sign_keypair(pk, sk);
+}
+
+int sphincs_plus_generate_keypair_from_seed(uint8_t *pk, uint8_t *sk,
+                                            const uint8_t *seed) {
+  return crypto_sign_seed_keypair(pk, sk, seed);
+}
+
+int sphincs_plus_sign(const uint8_t *message, uint32_t message_size,
+                      const uint8_t *sk, uint8_t *out_sign) {
+  size_t out_sign_len = 0;
+
+  int ret =
+      crypto_sign_signature(out_sign, &out_sign_len, message, message_size, sk);
+  if (ret != 0) {
+    return ret;
```

Comment:
You can return a more meaningful custom error code, consistent with the style of `sphincs_plus_verify`

Reply:
Done